src/app/core/services/fss-search-filter.service.ts

```diff
@@ -88,10 +88,11 @@ export class FssSearchFilterService {
     for (let i in searchKeywords) {
       if (searchKeywords[i] !== "") {
         let searchKeyword = searchKeywords[i].replace(/'/g, "''");
+        let defaultSearchExpression = "($batchContains('" + searchKeyword + "') OR contains(filename, '" + searchKeyword + "'))";
         if (filterExpression === "")
-          filterExpression = "$batchContains('" + searchKeyword + "')";
+          filterExpression = defaultSearchExpression;
         else
-          filterExpression = "(" + (filterExpression.concat(" OR ")).concat("$batchContains('" + searchKeyword + "')") + ")";
+          filterExpression = "(" + (filterExpression.concat(" OR ")).concat(defaultSearchExpression) + ")";
```

Comment:
Can you check the expression in case of more than 2 values?

Reply:
yes, resolved.